heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -970,7 +971,11 @@ export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
 
         // Example:  @rushstack/heft-jest-plugin/path/to/file.js
         if (propertyValue.startsWith(PLUGIN_PACKAGE_NAME)) {
-          const restOfPath: string = path.normalize('./' + propertyValue.slice(PLUGIN_PACKAGE_NAME.length));
+          let restOfPath: string = path.posix.normalize(
+            './' + propertyValue.slice(PLUGIN_PACKAGE_NAME.length)
+          );
+          restOfPath = restOfPath.replace(/^(\.\/)?lib(\/|\\)/, 'lib-commonjs/');
+
           return path.join(PLUGIN_PACKAGE_FOLDER, restOfPath);
```

Comment:
Although it will be slower, we may wish to route it through `require.resolve` to get it to respect the `exports` field.